docs/4.2. Behaviour - Querying.md

```diff
@@ -2,63 +2,63 @@
 
-There is no mandated URL base path for servers to use when creating websocket connections. Instead clients should observe the value of 'ws_href' which is returned by their subscription request in order to identify what to connect to.
+There is no mandated URL base path for servers to use when creating WebSocket connections. Instead clients SHOULD observe the value of 'ws_href' which is returned by their subscription request in order to identify what to connect to.
 
-Websockets created for specific clients will be cleaned up automatically if they disconnect.
+WebSockets created for specific clients will be cleaned up automatically if they disconnect.
 
-Connection upgrade procedures permitting an existing HTTP query to transition to a websocket are not currently supported, but may be in the future.
+Connection upgrade procedures permitting an existing HTTP query to transition to a WebSocket are not currently supported, but could be in the future.
 
 ### Query API Behaviour
-Long-lived connections to the Query API are supported via Websockets which can be set up via the /subscriptions resource. Query APIs may additionally support the HTTP 'Upgrade' header sent by clients to upgrade an HTTP GET request to a Websocket. In cases where this is performed, a corresponding entry in /subscriptions must also be created with matching query parameters.
 
-Upon receiving a request for a new subscription, the Query API should identify whether it already has any websockets open which provide for the requested query. If a websocket exists, the 'subscription' object representing it may be returned to the user. If a relevant websocket does not exist a new one should be created.
+Long-lived connections to the Query API are supported via WebSockets which can be set up via the /subscriptions resource. Query APIs MAY additionally support the HTTP 'Upgrade' header sent by clients to upgrade an HTTP GET request to a WebSocket. In cases where this is performed, a corresponding entry in /subscriptions MUST also be created with matching query parameters.
 
-In normal operation, once a websocket has no more clients subscribed to it the Query API may automatically remove it and its corresponding HTTP-advertised subscription object. The Query API MUST NOT acknowledge HTTP DELETE requests for websockets running in this 'non-persistent' mode, instead issuing an HTTP 403 response.
+Upon receiving a request for a new subscription, the Query API SHOULD identify whether it already has any WebSockets open which provide for the requested query. If a WebSocket exists, the 'subscription' object representing it MAY be returned to the user. If a relevant WebSocket does not exist a new one SHOULD be created.
```

Comment:
What does having a WebSocket "open" mean? Not a connection from a client, just whether or not it's currently listening for connections. Would be clearer if it just talked about having an appropriate subscription object...

Reply:
Would be better resolved through #162